src/transport/SecureSessionMgr.cpp

```diff
@@ -271,16 +272,19 @@ void SecureSessionMgrBase::HandleDataReceived(const PacketHeader & packetHeader,
 #endif
         plainText = msg->Start();
 
-        // TODO: We need length checks here!  https://github.com/project-chip/connectedhomeip/issues/2928
-        err = mac.Decode(packetHeader, &data[packetHeader.GetPayloadLength()], kMaxTagLen, &taglen);
+        payloadlen = packetHeader.GetPayloadLength();
+        VerifyOrExit(
+            payloadlen <= len,
```

Comment:
any reason why it `<` in the other file?

`VerifyOrExit(payloadlen < len, err = CHIP_ERROR_INCORRECT_STATE);`

Reply:
Er, yes.  I wrote them both as `<` first, then realized that `<=` might make more sense and failed to switch both.  Will fix that to be `<=` both places.